### PR TITLE
Give a good error message if the called for xmatter is missing

### PR DESCRIPTION
**Pull request by hatton**
This PR also makes the 2 messages encountered in this scenario localizable, and makes the edit and publish tabs be disabled if the select book has this kind of error. Previously, that happened only if the book's html could not validate.

### CODE REVIEW

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-debug/100/
Build succeeded

File: src/BloomExe/Book/Book.cs [hunk 1]
```diff
@@ -423,14 +424,21 @@ private HtmlDom GetBookDomWithStyleSheets(params string[] cssFileNames)
...
 			var builder = new StringBuilder();
-			builder.Append("<html><body>");
-			builder.AppendLine("<p>This book (" + StoragePageFolder + ") has errors.");
-			builder.AppendLine(
-				"This doesn't mean your work is lost, but it does mean that something is out of date or has gone wrong, and that someone needs to find and fix the problem (and your book).</p>");
+			builder.Append("<html><body style='font-family:arial,sans'>");
 
+			builder.AppendLine("<p>" + GetGenericBrokenBookRecommendationString() + "</p>");
+
+			builder.AppendFormat("<p>" + _storage.ErrorMessages + "</p>");
```
Comment (JohnThomson) [inline src/BloomExe/Book/Book.cs]: Are these three (I think) duplicated lines worth a method?

File: src/BloomExe/Book/BookStorage.cs [hunk 2]
```diff
@@ -656,9 +658,16 @@ private void UpdateSupportFiles()
...
 			var nameOfXMatterPack = _dom.GetMetaValue("xMatter", _collectionSettings.XMatterPackName);
-			var helper = new XMatterHelper(_dom, nameOfXMatterPack, _fileLocator);
-			UpdateIfNewer(Path.GetFileName(helper.PathToStyleSheetForPaperAndOrientation), helper.PathToStyleSheetForPaperAndOrientation);
 
+			try
+			{
+				var helper = new XMatterHelper(_dom, nameOfXMatterPack, _fileLocator);
+				UpdateIfNewer(Path.GetFileName(helper.PathToStyleSheetForPaperAndOrientation), helper.PathToStyleSheetForPaperAndOrientation);
+			}
+			catch (Exception error)
```
Comment (JohnThomson) [inline src/BloomExe/Book/BookStorage.cs]: Could we reasonably distinguish errors we should report like this (FileNotFound, IOException, maybe some to do with invalid XML) from more serious problems that are definitely bugs, like NullReference, KeyNotFound, OutOfMemory,...I know this is often hard and can live with this if you say so, but I (and MS guidelines) consider catching Exception (without differentiation) a code smell. I'm trying to resist doing it more than we already do.

File: src/BloomExe/Book/XMatterHelper.cs [hunk 3]
```diff
@@ -42,8 +45,13 @@ public XMatterHelper(HtmlDom bookDom, string nameOfXMatterPack, IFileLocator fil
 			}
 			catch (Exception error)
 			{
+				var errorTemplate = LocalizationManager.GetString("Errors.XMatterNotFound",
+					"This Book called for Front/Back Matter pack named '{0}', but Bloom couldn't find that on this computer. You can either install a BloomPack that will give you '{0}', or go to Settings:Book Making and change to another Front/Back Matter Pack.");
```
Comment (JohnThomson) [inline src/BloomExe/Book/XMatterHelper.cs]: Now this message is very specific to File Not Found, so I'd say it should definitely be limited to that exception.

File: src/BloomExe/Book/Book.cs (hunk 1 above)
Comment (hatton) [inline src/BloomExe/Book/Book.cs]: Asked myself the same question, decided they aren't :-)

File: src/BloomExe/Book/BookStorage.cs (hunk 2 above)
Comment (hatton) [inline src/BloomExe/Book/BookStorage.cs]: John, I appreciate you point. But in this case, it seems too weak to warrant trying to come up with this list of OK errors versus Unexpected errors. Note,  the catch here doesn't swallow anything. Rather, it surfaces the error to the user in a way that works even when loading up dozens of books, any of which may have problems. The ErrorMessages property that is set here is used to do a couple things, including displaying the error contents where the book preview would normally be, and disabling editing of the book.

File: src/BloomExe/Book/XMatterHelper.cs (hunk 3 above)
Comment (hatton) [inline src/BloomExe/Book/XMatterHelper.cs]: LocateDirectoryWithThrow throws an ApplicationException. I've changed it to only catch that.

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-debug/173/
Test PASSed.

**Comment by JohnThomson** [comment]
Fair enough...I'm willing for this to be merged if you don't plan any more changes.

**Pull request merged**